executor/aggfuncs/func_count_distinct.go

```diff
@@ -85,18 +95,28 @@ func (e *countOriginalWithDistinct4Int) UpdatePartialResult(sctx sessionctx.Cont
 		if isNull {
 			continue
 		}
-		if p.valSet.Exist(input) {
-			continue
+		if p.needSync {
```

Comment:
Is this branch always true?

Reply:
It is true only for HashAgg with distinct.

Reply:
The aggFunc is `countOriginalWithDistinct4Int`.

Reply:
``` mysql
create table t (a int, b int);
select /*+ stream_agg() */ count(distinct case when 1 then a else 0 end) from t;
```

For example, streamAgg will use `countOriginalWithDistinct4Int` but will not call `SetPartialResultAsNeedSync`, so that `p.needSync` is false.